prover/src/monolith/constraints/evaluation_table.rs

```diff
@@ -288,38 +275,162 @@ fn apply_divisor<B: StarkField, E: FieldElement + From<B>>(
         "multiple exclusion points are not yet supported"
     );
 
-    // convert the polynomial into coefficient form by interpolating the evaluations
-    // over the evaluation domain
-    fft::interpolate_poly_with_offset(&mut column, &inv_twiddles, domain_offset);
-    let mut poly = column;
+    // compute inverse evaluations of the divisor's numerator, which has the form (x^a - b)
+    let domain_size = column.len();
+    let z = get_inv_evaluation(divisor, domain_size, domain_offset);
 
-    // divide the polynomial by its divisor
-    let numerator = numerator[0];
-    let degree = numerator.0;
+    const MIN_CONCURRENT_SIZE: usize = 1024;
 
+    // divide column values by the divisor; for boundary constraints this computed simply as
+    // multiplication of column value by the inverse of divisor numerator; for transition
+    // constraints, it is computed similarly, but the result is also multiplied by the divisor's
+    // denominator (exclusion point).
     if divisor.exclude().is_empty() {
-        // the form of the divisor is just (x^degree - a)
-        let a = E::from(numerator.1);
-        polynom::syn_div_in_place(&mut poly, degree, a);
+        // the column represents merged evaluations of boundary constraints, and divisor has the
+        // form of (x^a - b); thus to divide the column by the divisor, we compute: value * z,
+        // where z = 1 / (x^a - 1) and has already been computed above.
+
+        if cfg!(feature = "concurrent") && result.len() >= MIN_CONCURRENT_SIZE {
+            #[cfg(feature = "concurrent")]
+            {
+                #[rustfmt::skip]
+                result.par_iter_mut().zip(column).enumerate().for_each(|(i, (result, value))| {
```

Comment:
Reusing `result` for `result_i` was a bit confusing for the reader, maybe rename to `result_i` or `col_i` etc.

Reply:
fixed!